lib/fennec/evaluator/allocate/request.ex

```diff
@@ -85,12 +85,14 @@ defmodule Fennec.Evaluator.Allocate.Request do
   end
 
   defp verify_unknown_attributes(params, state) do
-    case Params.get_attrs(params) do
-      [] ->
-        {:continue, params, state}
+    with u  <- Params.get_attr(params, Attribute.Username),
+         r  <- Params.get_attr(params, Attribute.Realm),
+         [] <- Params.get_attrs(params) -- [u, r] do
+      {:continue, params, state}
+    else
       _ ->
         {:error, %Attribute.ErrorCode{code: 420}}
```

Comment:
Yes, it should be added to the decoder. Unknown attributes should be collected throughout decoding process and presented somehow to the caller.